Cirrus CI: add Debian 10, FreeBSD 10/11/13

qris opened the pull request.

qris committed:
Cirrus CI: add Debian 10, FreeBSD 10/11/13

jamesog commented on .cirrus.yml:
```diff
@@ -58,13 +76,31 @@ task:
-    <<: *standard_gce_instance
-    image_project: freebsd-org-cloud-dev
-    image_name: freebsd-12-0-release-amd64
-    disk: 30  # Image is 29GB, VM can't be smaller
```
`disk` can't be dropped as the FreeBSD image has a larger disk. I guess we could make it larger in the `standard_gce_instance` but otherwise we need to include this on all FreeBSD instances.

jamesog commented on .cirrus.yml:
```diff
@@ -58,13 +76,31 @@ task:
+        <<: *standard_gce_instance
+        platform: FreeBSD
+        image_project: freebsd-org-cloud-dev
+        image_family: freebsd-13-0-snap
```
I think this needs to be `image_name`

jamesog commented on .cirrus.yml:
```diff
@@ -57,14 +75,31 @@ task:
+  image_project: freebsd-org-cloud-dev
+  # Need to use image_name for now: https://github.com/cirruslabs/cirrus-ci-docs/issues/422
+  disk: 30  # Image is 29GB, VM can't be smaller
+  <<: *standard_gce_instance
```
This needs to come earlier, otherwise `disk` from `standard_gce_instance` will overwrite the value above.

qris replied:
Thanks :)

jamesog reviewed (commented):
Just that last thing, otherwise LGTM.

jamesog commented:
Oh, I see all the Debian builds are failing now as it doesn't like the image name. Maybe it really does need `-stretch-v20190618` etc.

jamesog commented:
cirruslabs/cirrus-ci-docs#422

qris committed:
test_ssl_keepalives: relax timing tolerance for Cirrus CI

jamesog commented on .cirrus.yml:
```diff
@@ -38,17 +38,41 @@ task:
+      gce_instance:
+        <<: *standard_gce_instance
+        image_project: debian-cloud
+        image_name: debian-10-buster-v20190813
```
And this can become `image_family: debian-10`.

qris committed:
Merge branch 'master' into add_cirrus_ci_freebsd

qris committed:
Cirrus CI: replace hard-coded image names with image families

Remove obsolete distros and add newly available ones

qris committed:
Cirrus CI: stop trying to install libeditline-dev

It's been removed from Debian Stable (and thus Ubuntu), unfortunately:
https://packages.qa.debian.org/e/editline/news/20201108T213808Z.html

qris committed:
Merge branch 'master' into add_cirrus_ci_freebsd

qris committed:
Cirrus CI: hopefully fix Ubuntu builds

Remove 21.04 as it's no longer available non-deprecated on GCE. Change
18.04 image to Ubuntu Pro, as that still seems to be available.

qris committed:
Cirrus CI: Add Ubuntu 16.04 Pro LTS
